pkg/controllers/dynakube/kspm/daemonset/reconciler.go

```diff
@@ -0,0 +1,141 @@
+package daemonset
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/conditions"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/hasher"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/daemonset"
+	k8slabels "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/labels"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/node"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	nameSuffix         = "-node-config-collector"
+	serviceAccountName = "dynatrace-node-config-collector"
+)
+
+type Reconciler struct {
+	client    client.Client
+	apiReader client.Reader
+	dk        *dynakube.DynaKube
+}
+
+func NewReconciler(clt client.Client,
+	apiReader client.Reader,
+	dk *dynakube.DynaKube) *Reconciler {
+	return &Reconciler{
+		client:    clt,
+		apiReader: apiReader,
+		dk:        dk,
+	}
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context) error {
+	if !r.dk.KSPM().IsEnabled() {
+		if meta.FindStatusCondition(*r.dk.Conditions(), conditionType) == nil {
+			return nil // no condition == nothing is there to clean up
+		}
+
+		query := daemonset.Query(r.client, r.apiReader, log)
+		err := query.Delete(ctx, &appsv1.DaemonSet{ObjectMeta: metav1.ObjectMeta{Name: r.dk.KSPM().GetDaemonSetName(), Namespace: r.dk.Namespace}})
+
+		if err != nil {
+			log.Error(err, "failed to clean-up KSPM config-secret")
+		}
+
+		meta.RemoveStatusCondition(r.dk.Conditions(), conditionType)
+
+		return nil // clean-up shouldn't cause a failure
+	}
+
+	ds, err := r.generateDaemonSet()
+	if err != nil {
+		return err
+	}
+
+	updated, err := daemonset.Query(r.client, r.apiReader, log).WithOwner(r.dk).CreateOrUpdate(ctx, ds)
+	if err != nil {
+		conditions.SetKubeApiError(r.dk.Conditions(), conditionType, err)
+
+		return err
+	}
+
+	if updated {
+		conditions.SetDaemonSetOutdated(r.dk.Conditions(), conditionType, r.dk.KSPM().GetDaemonSetName()) // needed to reset the timestamp
+		conditions.SetDaemonSetCreated(r.dk.Conditions(), conditionType, r.dk.KSPM().GetDaemonSetName())
```

Comment:
What if we create a func in `conditions` package to update the timestamp instead? This looks a bit like a gimmick

Reply:
it's fine as is now, we can think about it in general how we want to handle this (will create a follow up)